### ISSUE TITLE
Failed to boot after update to 21.2 

### ISSUE THREAD

**Issue by QkiZMR** [topjohnwu/Magisk#3668]
Moto Z3 Play here, Android 9. Failed to boot after update from 21.0 to 21.2 after direct update from Magisk Manager. I can't upload magisk.log file because I can't boot system to copy it. I only get logcat from failed booting.
[output.zip](https://github.com/topjohnwu/Magisk/files/5798355/output.zip)
I tried 21.1 version but same error as in 21.2.

**Comment by linuxdep** [comment]
Hi, tghe same on HTC U11, black screen and only after akko 0% i can reboot to recovery.
20.4 works

**Comment by selurvedu** [comment]
Same here. I was able to reboot into recovery right away without having to drain the battery. After that I flashed the clean kernel and Magisk 20.4, it works as usual.

See also #3662. #3663 may also be related.

**Comment by ayanamist** [comment]
The same here on Xiaomi 10 (umi) China with miui_UMI_V12.0.11.0.QJBCNXM (android 10), after upgrade, it always boot loop, and there is no working recovery to mount encrypted data partition so i can't provide logs. All things were good when it is using magisk 20.4

**Comment by GiantWaffle** [comment]
Hi,

Same here on my LG (H910A) V20 running 7.0 and it crashes to a gray screen (too tiny to read).  So I used TWRP to restore to 20.4

I saw an option that I never saw before when I went to update and I was wondering if my having failed to check that box when it gave me the choice is why it didn't work.  The option was;

Preserve AVB 2.0/dm-verity

I don't know what that is, but could that be why it didn't work?  Should I try checking that box?

Thanks!

**Comment by ZeCracotte** [comment]
Same problem here on Asus Zenfone 3 ZE552KL : I was stuck on a bootloop, I went back to v20.4 and it worked again.

**Comment by jenaplisken** [comment]
@ayanamist thanks. With your trick i patched boot.img of EEA 12.0.18 out today.
Waiting for a solution for magisk 21.2

**Comment by AymaneShadow** [comment]
Same problem here. I used the Magisk Manager to directly upgrade to version 21.2 on Samsung Galaxy S6 Edge with Android version 7.0, it says that it installed it correctly and after tapping the Reboot button the device going into a bootloop.

**Comment by doclucas** [comment]
Same problem on Galaxy note 10+ N975F (Android 10)
Booting into TWRP and flashing 20.4 doesn't fix the stuck boot for me.
Only uninstalling Magisk make the phone boot now. :(

**Comment by dontobi** [comment]
I have the same problem with my Xiaomi Mi 10 (Xiaomi.eu Rom 12.2.8.0 - Android 11). When I flash Magisk v21.2, my device gets stuck in the bootloader. On the other hand, when I flash v21.1, my device boots without problems.
I also tried the Canary version, but it does not boot either.
I can rule out the modules used as the cause of the problem, because I use the same ones on a Xiaomi Mi Pad 4 (PixelExperience Plus Rom - Android 10).

**Comment by AymaneShadow** [comment]
> 
> 
> Same problem on Galaxy note 10+ N975F (Android 10)
> Booting into TWRP and flashing 20.4 doesn't fix the stuck boot for me.
> Only uninstalling Magisk make the phone boot now. :(

May I ask you, how did you uninstall Magisk?

**Comment by v1993** [comment]
@AymaneShadow You can download magisk uninstaller in "Realeases" tab of this repository, then flash it with TWRP.

**Comment by theft07** [comment]
same here. after updating from 20.4 stable to 21.2 stable inside manager, my huawei y6 (2018) refuses to boot. if i revert back to 20.4 in twrp the phone boots fine.

**Comment by dlohuis** [comment]
Same issue here on my Blackview BV9900 Pro.
-> Magisk Manager
-> Current version 20.4 (20400), New version 21.2 (21200), Update
-> Both options Checked, Next
-> Direct Install (Recommended)
-> Boot Loop

After flashing the recovery image I still had Device boots normal again with Magisk version 20.4

**Comment by Korel** [comment]
Same here, Samsung Galaxy S6 SM-G920F. Updated using Magisk manager update, 
* Preserve AVB 2.0/dm-verity checked. 
* Direct Install (Recommended)
After boot, directly boots to TWRP and only TWRP. Uninstalled Magisk from TWRP and reinstalled version 21.0. Boots normally again.

**Comment by Snaxeko** [comment]
Same here, Realme x50 pro (rmx2075). Doesn't boot, tried canary and stable.

**Comment by topjohnwu** [comment]
I just released a new canary build, may you please try it out and report back?

**Comment by dlohuis** [comment]
@topjohnwu

I installed the new canary build with success.

![Screenshot_20210114-190614](https://user-images.githubusercontent.com/15121025/104631133-190b7400-569c-11eb-8256-7f3b7da4d5f9.png)

👍

**Comment by dontobi** [comment]
> I just released a new canary build, may you please try it out and report back?

I have just tested the latest Canary version. Performed an update via the Magisk Manager app and sent my Mi 10 (umi - Xiaomi.eu 12.2.8.0) into a reboot. Unfortunately it still gets stuck in the bootloader.

I then booted TWRP, flashed the Xiaomi.eu Rom and then Magisk Canary, but again my device got stuck in the bootloader.

The last thing I did was to flash the Uninstaller Zip via TWRP and then Magisk Canary. Now my device has booted. To be on the safe side, I sent it into a reboot again and this also worked.
Then I installed the Busybox module and rebooted again. This also worked.
Finally, my own module came to the turn that my Mi10 also adapts my needs.  Unfortunately, my device now no longer booted. Thus, in my case, it seems to be due to my own module. However, I do not know how I can troubleshoot to find out what is wrong with it. Because on my MiPad 4 runs the same module with Magisk 21.2 and also with the Canary versions.
I will now go through my module and operate trial and error. I don't know what else to do at the moment.

**Comment by theft07** [comment]
Flashing the 21202 zip inside the manager with direct install, with and without the option preserve 'AVB 2.0/dm-verity' checked, my huawei y6 (2018)
(Ramdisk Yes
A/B No
SAR No
Crypto File)
still bootloops. I guess I'll have to stay on 20.4...

**Comment by blackbox87** [comment]
> I have the same problem with my Xiaomi Mi 10 (Xiaomi.eu Rom 12.2.8.0 - Android 11). When I flash Magisk v21.2, my device gets stuck in the bootloader.

I tried the 76061296 version that was uploaded today and experienced the exact same issue with an Xiaomi Poco F1 running LineageOS 16. My phone goes imminently to the bootloader and remains there unless I boot to recovery and flash Magisk v21.2.

**Comment by QkiZMR** [comment]
> I just released a new canary build, may you please try it out and report back?

Still the same error. 
[logcat.zip](https://github.com/topjohnwu/Magisk/files/5817246/logcat.zip)
Magisk canary.
`Magisk 76061296(21202) daemon started`
Motorola Moto Z3 Play, stock Pie.

**Comment by auroszx** [comment]
Had a similar issue today with my Nexus 5X (running LineageOS 17.1) after updating to build 76061296. Rather than bootlooping the first boot got really slow (as in, the bootanimation was barely moving), but after rebooting by holding the power button it actually worked just fine. Not sure what's up with that.

**Comment by QkiZMR** [comment]
Installing the canary version completely broke my Magisk installation. I removed Magisk with uninstaller and now I'm trying to install version 21.2 from scratch. Maybe it helps.

**Comment by blackbox87** [comment]
@topjohnwu [My issue](https://github.com/topjohnwu/Magisk/issues/3668#issuecomment-760470797) is caused by c91c070.

I compiled Magisk from 7606129 and kept reverting changes until my phone was booting again. So for me the last working build is aec06a6.

**Comment by doclucas** [comment]
[**@doclucas**](https://github.com/doclucas) commented on [Jan 13, 2021, 12:56 AM GMT+7](https://github.com/topjohnwu/Magisk/issues/3668#issuecomment-758832999 "2021-01-12T17:56:31Z - Replied by Github Reply Comments"):
> Same problem on Galaxy note 10+ N975F (Android 10)
> Booting into TWRP and flashing 20.4 doesn't fix the stuck boot for me.
> Only uninstalling Magisk make the phone boot now. :(

I think it's worth noting that in my case, the rom has been pre-patched by magisk manager to be rooted so I am on 20.4 since day one. Also trying to install 20.4 on this ROM make it no longer boot until I uninstall magisk via TWRP (which keeps is still rooted and working fine after booting and installing Magisk manager).
So I am assuming that in my specific case I can't even upgrade magisk unless I prepatch the original stock rom again with the latest manager, right?

**Comment by topjohnwu** [comment]
I just released yet another canary build, it should fix bootloops for many devices. If you still experience bootloops, open a new issue with **proper information** needed for debugging. Thank you!
